test/unit_test/static/quantity_test.cpp

```diff
@@ -234,6 +234,18 @@ static_assert(length<kilometre, int>(2_q_km).number() == 2);
 static_assert(length<metre, int>(2_q_km).number() == 2000);
 static_assert(length<kilometre>(1500_q_m).number() == 1.5);
 
+///////////////////////////////////////
+// derived quantities
+///////////////////////////////////////
+
+template<Representation Rep, units::Quantity Q, const units::basic_fixed_string additional_nttp_argument>
+struct derived_quantity : quantity<typename Q::dimension, typename Q::unit, Rep> {
```

Comment:
So we have:

```cpp
derived_quantity<double, si::length<si::metre>, "a"> q1;
derived_quantity<double, si::length<si::kilometre>, "b"> q2;
auto q = q1 + q2;
```

The PR description suggests that the inheritance is selected over composition because one may want to reuse the operators. However `q1 + q2` will return some kind of a `quantity<...>` rather than `derived_quantity<...>` which probably is not intended. As a result, all of the operators have to be redefined for `derived_quantity` as well to properly handle the return type, right?

Reply:
> [..] which probably is not intended
Actually yes, that was our intention ... at least that is in our use-case. As mentioned above, the unit-test is more functional and doesn't convey that aspect of the intention.

We'd primarily like to inherit the `Quantity` concept and math operator overloads. The rest of the code would otherwise use standard 'quantity' definitions. I didn't want to make a specific extension that would work only/primarily for our use-case but tried to keep it a bit more generic for other users and/or our other uses-cases.

N.B. for what it's worth, a bit more detail on our use case: we add additional NTTP meta-information to the derived quantity template that are not directly linked to the quantity but rather to IO operations, static reflection ((de-)serialising), and external access rules (context: micro-services) such as:
  * documenting the data struct member (ie. [OpenAPI](https://github.com/OAI/OpenAPI-Specification) meta-infos),
  * define external read/write access definition (e.g. internally non-const/constexpr -- since service needs to modify/recompute them -- but access from external function/users may be restricted), 
  * relationship between different member fields (ie. whether external access may modify individual fields, only in groups, or the whole data structure), and
  * Role-Based-Access ([RBAC](https://csrc.nist.gov/projects/role-based-access-control)) rules.

If you derive new quantities through, for example `q1 + q2`, most of these IO meta-infos -- notably the variable 'description' -- become irrelevant until the final value in the processing chain ist assigned to a new 'Annotated' quantity that can be (de-)serialised.

Our present signature looks a bit like (assuming the new quantity extension):

```cpp
template<units::Representation Rep, units::Unit U, const StringLiteral description = "", 
    const IoAccess direction = BOTH, const IoGroup group = NONE, const RbacRole = ALL>
struct Annotated : public units::quantity<U, Rep> { 
 /* [...] */ 
}
```
Most of these additional NTTPs are constexpr (or at least defined during compile-time) but may change their signature...

Reply:
@JohelEGP `Annotated` is basically a `quantity` that is annotated for IO purposes. We need both the NTTPs as well as the mathematical operations that are defined for `Quanitty`. Through inheritance one basically inherits a perfect operator forwarding rather than having to rewrite all operators/cases/exceptions by hand.

For example, the data flow in our microservice applications is typically:
A) net-IO generates and verifies a new input domain object -- ie. data class containing multiple `Annotated<...>` fields,
B) multiple modules perform post-/internal-processing using the input domain objects and other regular `quantity`, and
C) result is copied to an output domain object -- also containing multiple `Annotated<...>` fields -- and serialised to IO.

In step 'A)' the additional NTTPs are used to validate the external user input (received via network), and in step 'C)' do verify and generate the OpenAPI documentation. We trying to keep the data and behaviour definitions together. 

The 'Annotated` template helps us in this respect that all the relevant information has to be defined only once and in a single location (ie. domain object class/struct definition). The input/output validation is basically the conceptual extension of the 'units' physical unit safety concept to network IO-API safety...

Hope this helps. Let me know if you have further questions, suggestions or recommendations.

Reply:
OK. So the problem is that by defining the arithmetic and IO operators in terms on `Quantity` makes it so that they don't work on `Annotated` arguments, as opposed to parameters taking a specialization of `quantity`, in which case `Annotated` would convert to its base `quantity`.

Reply:
Basically, yes. I tried without inheritance but then forwarding the operators became quite unwieldy.